doc/api/modules.md

```diff
@@ -600,22 +602,15 @@ Process files with the extension `.sjs` as `.js`:
 require.extensions['.sjs'] = require.extensions['.js'];
 ```
 
-**Deprecated** In the past, this list has been used to load
-non-JavaScript modules into Node.js by compiling them on-demand.
-However, in practice, there are much better ways to do this, such as
-loading modules via some other Node.js program, or compiling them to
-JavaScript ahead of time.
-
-Since the module system is locked, this feature will probably never go
-away. However, it may have subtle bugs and complexities that are best
-left untouched.
-
-Note that the number of file system operations that the module system
-has to perform in order to resolve a `require(...)` statement to a
-filename scales linearly with the number of registered extensions.
+**Deprecated** In the past, this list has been used to load non-JavaScript
+modules into Node.js by compiling them on-demand. However, in practice, there
+are much better ways to do this, such as loading modules via some other Node.js
+program, or compiling them to JavaScript ahead of time.
 
-In other words, adding extensions slows down the module loader and
-should be discouraged.
+Using this could cause subtle bugs and adding extensions slows down the module
+loader and is therefore discouraged (the number of file system operations that
+the module system has to perform in order to resolve a `require(...)` statement
+to a filename scales linearly with the number of registered extensions).
```

Comment:
This is a very long sentence that would likely be more clear if it were split up into shorter statements. Perhaps the parenthetical explanation is unnecessary? Here's a suggestion:

```markdown
Avoid using `require.extensions`. Use could cause subtle bugs. Adding extensions
slows down the module loader.
```

Reply:
It actually does not slow down any other calls due to the execution order of the calls (maybe slightly due to an increase in cached entries but that's not too bad). Only newly added extensions are loaded slower. If a bunch is added, they slow each other down and that's why I tried to keep some information about that.